propagate_type_hints.py

```diff
@@ -0,0 +1,111 @@
+# Propagate type hints & signatures defined in _linear_operator.py to derived classes.
```

Comment:
Maybe add the usage of this script to the `CONTRIBUTING.md` guide along with any other notes on typing

Reply:
Good idea.